wdl-gauntlet/src/report.rs

```diff
@@ -16,14 +16,14 @@ pub struct UnmatchedStatus {
     ///
     /// These are the diagnostics that were in the config but were not
     /// emitted by the parser/validator.
-    pub missing: IndexSet<String>,
+    pub missing: IndexSet<(Option<usize>, String)>,
     /// The unexpected set of diagnostics.
     ///
     /// These are the diagnostics that were not in the configuration but
     /// were emitted by the parser/validator.
-    pub unexpected: IndexSet<String>,
+    pub unexpected: IndexSet<(Option<usize>, String)>,
     /// The set of all diagnostics that were emitted.
-    pub all: IndexSet<String>,
+    pub all: IndexSet<(Option<usize>, String)>,
```

Comment:
Maybe only `all` needs to be `IndexMap<String, Option<usize>>` and `missing` and `unexpected` can remain as they are?

That means we need to manually produce `missing` and `unexpected` rather than using the `-` operator on the sets, but that shouldn't be too tricky.